hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4239,7 +4242,17 @@ func (r *HostedClusterReconciler) validateAzureConfig(ctx context.Context, hc *h
 		}
 	}
 
-	return utilerrors.NewAggregate(errs)
+	if len(errs) > 0 {
```

Comment:
why not include the error below in all the validation errors?

Reply:
It looked to me that was particularly tied to the `for` loop on L4239. 

I don't have any issues adding it to that though.